src/pages/task/taskTabs/buildBaronAndAnnotations/AddIssueModal/index.tsx

```diff
@@ -0,0 +1,136 @@
+import { useMutation } from "@apollo/client";
+import Button from "@leafygreen-ui/button";
+import TextInput from "@leafygreen-ui/text-input";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { useAnnotationAnalytics } from "analytics";
+import { Accordion } from "components/Accordion";
+import { Modal } from "components/Modal";
+import { useToastContext } from "context/toast";
+import {
+  AddAnnotationIssueMutation,
+  AddAnnotationIssueMutationVariables,
+  IssueLinkInput,
+} from "gql/generated/types";
+import { ADD_ANNOTATION } from "gql/mutations";
+import { useAddIssueModal } from "./state";
+import { toDecimal } from "./utils";
+
+interface Props {
+  visible: boolean;
+  dataCy: string;
+  closeModal: () => void;
+  setSelectedRowKey: (key: string) => void;
+  taskId: string;
+  execution: number;
+  isIssue: boolean;
+}
+
+export const AddIssueModal: React.VFC<Props> = ({
+  visible,
+  dataCy,
+  closeModal,
+  setSelectedRowKey,
+  taskId,
+  execution,
+  isIssue,
+}) => {
+  const annotationAnalytics = useAnnotationAnalytics();
+  const dispatchToast = useToastContext();
+  const title = isIssue ? "Add Issue" : "Add Suspected Issue";
+  const issueString = isIssue ? "issue" : "suspected issue";
+  const analyticsType = isIssue
+    ? "Add Task Annotation Issue"
+    : "Add Task Annotation Suspected Issue";
+
+  const [state, dispatch] = useAddIssueModal();
+
+  const [addAnnotation] = useMutation<
+    AddAnnotationIssueMutation,
+    AddAnnotationIssueMutationVariables
+  >(ADD_ANNOTATION, {
+    onCompleted: () => {
+      dispatchToast.success(`Successfully added ${issueString}`);
+      dispatch.reset();
+      setSelectedRowKey(state.issueKey);
+      closeModal();
+      annotationAnalytics.sendEvent({ name: analyticsType });
+    },
+    onError(error) {
+      dispatchToast.error(
+        `There was an error adding the issue: ${error.message}`
+      );
+    },
+    refetchQueries: ["GetAnnotationEventData"],
+  });
+
+  const handleSubmit = () => {
+    const apiIssue: IssueLinkInput = {
+      url: state.url,
+      issueKey: state.issueKey,
+      confidenceScore: toDecimal(state.confidenceScore),
+    };
+    addAnnotation({ variables: { taskId, execution, apiIssue, isIssue } });
+  };
+
+  const handleCancel = () => {
+    dispatch.reset();
+    closeModal();
+  };
+
+  return (
+    <Modal
+      data-cy={dataCy}
+      visible={visible}
```

Comment:
I think you can replace this `Modal` with the `ConfirmationModal` from `components/ConfirmationModel`. (I tried it myself and double checked that there were no modal issues, I think it should work!)

You'll have to forgo the tooltip on the button, but I'm pretty sure a disabled button would be self-explanatory.